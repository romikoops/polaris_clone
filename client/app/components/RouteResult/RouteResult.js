import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './RouteResult.scss';
import { moment } from '../../constants';
export class RouteResult extends Component {
    constructor(props) {
        super(props);
        console.log(this.props);
    }

    render() {
        const sched = this.props.schedule;
        const { theme } = this.props;
        const schedKey = sched.starthub_id + '-' + sched.endhub_id;
        const borderColour = theme && theme.colors ? '-webkit-linear-gradient(top, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'floralwhite';
        const borderStyle = {
            borderImage: borderColour
        };
        let originHub = {};
        let destHub = {};
        if (this.props.originHubs) {
            this.props.originHubs.forEach(hub =>  {
                if (hub.id === sched.starthub_id) {
                    originHub = hub;
                }
            });
            this.props.destinationHubs.forEach(hub =>  {
                if (hub.id === sched.endhub_id) {
                    destHub = hub;
                }
            });
        }


        return (
          <div className="flex-100 layout-row">
          <div className="flex-75 layout-row layout-wrap">
            <div className="flex-100 layout-row layout-align-start-center">
              <div className="flex-40 layout-row">
                <div className="flex-15 layout-column layout-align-start-center">
                  <i className="fa fa-map-marker"></i>
                </div>
                 <div className="flex-55 layout-row layout-wrap">
                  <h4 className="flex-100"> {originHub.name} </h4>
                </div>
                <div className="flex-100">
                  <p className="flex-100"> {originHub.code} </p>
                </div>
              </div>
              <div className="flex-15 layout-row layout-wrap layout-align-center-start" >
                <div className="flex-100 layout-row layout-align-center-center dash_border" style={borderStyle}>
                  <i className="fa fa-ship flex-none"></i>
                </div>
              </div>
              <div className="flex-40 layout-row">
                <div className="flex-15 layout-column layout-align-start-center">
                  <i className="fa fa-flag-o"></i>
                </div>
                <div className="flex-55 layout-row layout-wrap">
                  <h4 className="flex-100"> {destHub.name} </h4>
                </div>
                <div className="flex-100">
                  <p className="flex-100"> {destHub.code} </p>
                </div>
              </div>
            </div>
            <div className="flex-100 layout-row layout-align-start-center">
                <div className="flex-33 layout-wrap layout-row layout-align-center-center">
                  <div className="flex-100 layout-row">
                    <h4 className="flex-90">Pickup Date</h4>
                  </div>
                  <div className="flex-100 layout-row">
                    <p className="flex-50"> <strong> {moment(this.props.pickupDate).format('YYYY-MM-DD')} </strong></p>
                    <p className="flex-50"> {moment(this.props.pickupDate).format('HH:mm')}</p>
                  </div>

                </div>
                <div className="flex-33 layout-wrap layout-row layout-align-center-center">
                  <div className="flex-100 layout-row">
                    <h4 className="flex-90"> Date of Departure</h4>
                  </div>
                  <div className="flex-100 layout-row">
                    <p className="flex-50"> <strong> {moment(sched.etd).format('YYYY-MM-DD')}</strong></p>
                    <p className="flex-50"> {moment(sched.etd).format('HH:mm')}</p>
                  </div>

                </div>
                <div className="flex-33 layout-wrap layout-row layout-align-center-center">
                  <div className="flex-100 layout-row">
                    <h4 className="flex-90"> ETA terminal</h4>
                  </div>
                  <div className="flex-100 layout-row">
                    <p className="flex-50"> <strong> {moment(sched.eta).format('YYYY-MM-DD')}</strong></p>
                    <p className="flex-50"> {moment(sched.eta).format('HH:mm')}</p>
                  </div>

                </div>
            </div>
          </div>
          <div className="flex-25 layout-row layout-wrap">
            <div className="flex-100 layout-row layout-align-space-between-center layout-wrap">
              <p className="flex-none"> Per unit</p>
              <h4 className="flex-none"> {this.props.fees[schedKey].total.toFixed(2)} </h4>
            </div>
            <div className="flex-100 layout-row layout-align-space-between-center layout-wrap">
              <p className="flex-none"> Per unit</p>
              <h4 className="flex-none"> {this.props.fees[schedKey].total.toFixed(2)} </h4>
            </div>
          </div>
        </div>
        );
    }
}
RouteResult.PropTypes = {
    theme: PropTypes.object,
    schedule: PropTypes.object,
    selectResult: PropTypes.func,
    pickupDate: PropTypes.string,
    fees: PropTypes.object,
    originHubs: PropTypes.array,
    destinationHubs: PropTypes.array
};
