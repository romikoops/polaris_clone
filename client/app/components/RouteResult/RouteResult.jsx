import React, { Component } from 'react'
import PropTypes from '../../prop-types'
import styles from './RouteResult.scss'
import { moment } from '../../constants'
import { RoundButton } from '../RoundButton/RoundButton'
import { Price } from '../Price/Price'

export class RouteResult extends Component {
  static switchIcon (sched) {
    let icon
    switch (sched.mode_of_transport) {
      case 'ocean':
        icon = <i className="fa fa-ship" />
        break
      case 'air':
        icon = <i className="fa fa-plane" />
        break
      case 'train':
        icon = <i className="fa fa-train" />
        break
      default:
        icon = <i className="fa fa-ship" />
        break
    }
    return icon
  }
  static dashedGradient (color1, color2) {
    return `linear-gradient(to right, transparent 70%, white 30%), linear-gradient(to right, ${color1}, ${color2})`
  }
  constructor (props) {
    super(props)
    this.selectRoute = this.selectRoute.bind(this)
  }
  selectRoute () {
    const { schedule, fees } = this.props
    const schedKey = schedule.hub_route_key
    const totalFees = fees[schedKey].total
    this.props.selectResult({ schedule, total: totalFees })
  }
  render () {
    const {
      theme, schedule, user, pickup
    } = this.props
    const schedKey = schedule.hub_route_key
    const hubKeyArr = schedKey.split('-')
    let originHub = {}
    let destHub = {}
    if (this.props.originHubs) {
      this.props.originHubs.forEach((hub) => {
        if (String(hub.id) === hubKeyArr[0]) {
          originHub = hub
        }
      })
      this.props.destinationHubs.forEach((hub) => {
        if (String(hub.id) === hubKeyArr[1]) {
          destHub = hub
        }
      })
    }
    // const gradientFontStyle = {
    //     background:
    //         theme && theme.colors
    //             ? `-webkit-linear-gradient(left, ${
    //                 theme.colors.brightPrimary
    //             }, ${theme.colors.brightSecondary})`
    //             : 'black'
    // };
    const dashedLineStyles = {
      marginTop: '6px',
      height: '2px',
      width: '100%',
      background:
        theme && theme.colors
          ? RouteResult.dashedGradient(theme.colors.primary, theme.colors.secondary)
          : 'black',
      backgroundSize: '16px 2px, 100% 2px'
    }
    return (
      <div key={schedule.id} className={`flex-100 layout-row ${styles.route_result}`}>
        <div className="flex-75 layout-row layout-wrap">
          <div className={`flex-100 layout-row layout-align-start-center ${styles.top_row}`}>
            <div className={`flex-80 layout-row layout-align-start-center ${styles.hubs_row}`}>
              <div className={`${styles.header_hub}`}>
                <i className={`fa fa-map-marker ${styles.map_marker}`} />
                <div className="flex-100 layout-row">
                  <h4 className="flex-100"> {originHub.name} </h4>
                </div>
                {originHub.hub_code ? (
                  <div className="flex-100">
                    <p className="flex-100"> {originHub.hub_code}</p>
                  </div>
                ) : (
                  ''
                )}
              </div>
              <div className={`${styles.connection_graphics}`}>
                <div className="flex-none layout-row layout-align-center-center">
                  {RouteResult.switchIcon(schedule)}
                </div>
                <div style={dashedLineStyles} />
              </div>
              <div className={`${styles.header_hub}`}>
                <i className={`fa fa-flag-o ${styles.flag}`} />
                <div className="flex-100 layout-row">
                  <h4 className="flex-100"> {destHub.name} </h4>
                </div>
                <div className="flex-100">
                  <p className="flex-100"> {destHub.hub_code ? destHub.hub_code : ''} </p>
                </div>
              </div>
            </div>
            <div className={`flex-20 layout-row layout-align-start-center ${styles.load_type}`}>
              {/* <p className="flex-none no_m">{loadType}</p> */}
            </div>
          </div>
          <div className="flex-100 layout-row layout-align-start-center">
            <div className="flex-33 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}>{pickup ? 'Pickup Date' : 'Closing Date'}</h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(this.props.pickupDate).format('DD-MM-YYYY')}{' '}
                </p>
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(this.props.pickupDate).format('HH:mm')}{' '}
                </p>
              </div>
            </div>
            <div className="flex-33 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}> Date of Departure</h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(schedule.etd).format('DD-MM-YYYY')}{' '}
                </p>
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(schedule.etd).format('HH:mm')}{' '}
                </p>
              </div>
            </div>
            <div className="flex-33 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}> ETA terminal</h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(schedule.eta).format('DD-MM-YYYY')}{' '}
                </p>
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(schedule.eta).format('HH:mm')}{' '}
                </p>
              </div>
            </div>
          </div>
        </div>
        <div className="flex-25 layout-row layout-wrap">
          <div className="flex-100 layout-row layout-align-space-between-center layout-wrap">
            <p className="flex-none">Total price: </p>
            <Price value={this.props.fees[schedKey].total} user={user} />
          </div>
          <div className="flex-100 layout-row layout-align-space-between-center layout-wrap">
            <RoundButton
              text="Choose"
              size="full"
              handleNext={this.selectRoute}
              theme={theme}
              active
            />
          </div>
        </div>
      </div>
    )
  }
}
RouteResult.propTypes = {
  theme: PropTypes.theme,
  schedule: PropTypes.schedule.isRequired,
  selectResult: PropTypes.func.isRequired,
  pickupDate: PropTypes.string.isRequired,
  fees: PropTypes.objectOf(PropTypes.shape({
    total: PropTypes.any
  })).isRequired,
  originHubs: PropTypes.arrayOf(PropTypes.hub),
  destinationHubs: PropTypes.arrayOf(PropTypes.hub),
  user: PropTypes.user.isRequired,
  pickup: PropTypes.bool
}
RouteResult.defaultProps = {
  theme: null,
  pickup: false,
  originHubs: [],
  destinationHubs: []
}

export default RouteResult
