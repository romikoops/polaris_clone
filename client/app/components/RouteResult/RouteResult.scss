:local {
	.route_result {
		background-color: #FCFCFC;
		box-shadow: 0 2px 4px 0 rgba(0,0,0,0.17);
		margin: 10px 0;
		padding: 25px 36px;
		position: relative;
		h4, p {
			margin: 0;
		}
		h4 {
			font-weight: normal;
		}
	}
	.top_row {
		margin-bottom: 20px;
	}
	.header_hub {
		position: relative;
		.map_marker {
			position: absolute;
			left: -18px;
			top: -1px;
		}
		.flag {
			position: absolute;
			left: -23px;
			top: -1px;
		}
	}
	.connection_graphics {
		margin: -41px 27px 0 10px;
		width: 110px;
	}

	h4.date_title {
		-webkit-background-clip: text !important;
		-webkit-text-fill-color: transparent !important;
		text-shadow: 1px 1px rgba(black, .14);
		opacity: .7;
		margin-bottom: 8px;
		font-size: 14px;
	}
	p.sched_elem {
		margin-right: 10px;
		font-size: 14px;
	}
	.total_price {
		font-size: 30px;
		span {
			margin: 0 2px;
			font-size: 16px;
		}
		span.total_price_currency {
			font-size: 22px;
		}
	}
}