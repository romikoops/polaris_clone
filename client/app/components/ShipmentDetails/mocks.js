// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
