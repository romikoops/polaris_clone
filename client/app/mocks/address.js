export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
