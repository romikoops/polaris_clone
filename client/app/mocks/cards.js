export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
