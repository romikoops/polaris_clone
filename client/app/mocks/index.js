export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
