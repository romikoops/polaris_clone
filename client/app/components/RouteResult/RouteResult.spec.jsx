import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, user, hub, schedule, identity } from '../../mocks'

import RouteResult from './RouteResult'

const editedSchedule = {
  ...schedule,
  total_price: { currency: 'EUR', value: 112 },
  origin_hub: {},
  destination_hub: {},
  closing_date: 1522259422177,
  etd: 1522250422177,
  eta: 1522269422177
}
const propsBase = {
  theme,
  schedule: editedSchedule,
  selectResult: identity,
  fees: {
    [schedule.hub_route_key]: {
      total: { value: 12 }
    }
  },
  originHubs: [hub],
  destinationHubs: [hub],
  user,
  pickup: true
}

test('shallow rendering', () => {
  expect(shallow(<RouteResult {...propsBase} />)).toMatchSnapshot()
})

test('props.pickup is false', () => {
  const props = {
    ...propsBase,
    pickup: false
  }
  expect(shallow(<RouteResult {...props} />)).toMatchSnapshot()
})
