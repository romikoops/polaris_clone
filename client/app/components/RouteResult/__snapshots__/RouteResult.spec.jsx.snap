// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props.pickup is false 1`] = `
<div
  className="flex-100 layout-row route_result"
  key="555777"
>
  <div
    className="flex-70 layout-row layout-wrap"
  >
    <div
      className="flex-100 layout-row layout-align-start-center top_row"
    >
      <div
        className="flex-80 layout-row layout-align-start-center hubs_row"
      >
        <div
          className="header_hub"
        >
          <i
            className="fa fa-map-marker map_marker"
          />
          <h4
            className="flex-100"
          >
             
             
          </h4>
        </div>
        <div
          className="connection_graphics flex"
        >
          <div
            className="flex-none layout-row layout-align-center-center"
          >
            <i
              className="fa fa-anchor  "
              style={Object {}}
            />
          </div>
          <div
            style={
              Object {
                "background": "linear-gradient(to right, transparent 70%, white 30%), linear-gradient(to right, #333, #fafafa)",
                "backgroundSize": "16px 2px, 100% 2px",
                "height": "2px",
                "marginTop": "6px",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="header_hub"
        >
          <i
            className="fa fa-flag-o flag"
          />
          <h4
            className="flex-100"
          >
             
             
          </h4>
          <div
            className="flex-100"
          >
            <p
              className="flex-100"
            >
               
               
            </p>
          </div>
        </div>
      </div>
      <div
        className="flex-20 layout-row layout-wrap layout-align-start-center carrier_service_level"
      />
    </div>
    <div
      className="flex-100 layout-row layout-align-start-center"
    >
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            Closing Date
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            28-03-2018
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            ETD 
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            28-03-2018
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            ETA  
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            28-03-2018
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             Estimated T/T 
            <sup>
              *
            </sup>
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            0
             Days
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-30 layout-row layout-wrap layout-align-center"
  >
    <div
      className="flex-90 layout-row layout-align-space-between-center layout-wrap"
    >
      <p
        className="flex-none"
      >
        Total price: 
      </p>
      <Price
        currency="EUR"
        scale=""
        value={112}
      />
    </div>
    <div
      className="flex-90 layout-row layout-align-space-between-center layout-wrap"
    >
      <RoundButton
        active={true}
        back={false}
        disabled={false}
        handleDisabled={null}
        handleNext={[Function]}
        icon=""
        iconClass=""
        inverse={false}
        size="full"
        text="Choose"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`shallow rendering 1`] = `
<div
  className="flex-100 layout-row route_result"
  key="555777"
>
  <div
    className="flex-70 layout-row layout-wrap"
  >
    <div
      className="flex-100 layout-row layout-align-start-center top_row"
    >
      <div
        className="flex-80 layout-row layout-align-start-center hubs_row"
      >
        <div
          className="header_hub"
        >
          <i
            className="fa fa-map-marker map_marker"
          />
          <h4
            className="flex-100"
          >
             
             
          </h4>
        </div>
        <div
          className="connection_graphics flex"
        >
          <div
            className="flex-none layout-row layout-align-center-center"
          >
            <i
              className="fa fa-anchor  "
              style={Object {}}
            />
          </div>
          <div
            style={
              Object {
                "background": "linear-gradient(to right, transparent 70%, white 30%), linear-gradient(to right, #333, #fafafa)",
                "backgroundSize": "16px 2px, 100% 2px",
                "height": "2px",
                "marginTop": "6px",
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="header_hub"
        >
          <i
            className="fa fa-flag-o flag"
          />
          <h4
            className="flex-100"
          >
             
             
          </h4>
          <div
            className="flex-100"
          >
            <p
              className="flex-100"
            >
               
               
            </p>
          </div>
        </div>
      </div>
      <div
        className="flex-20 layout-row layout-wrap layout-align-start-center carrier_service_level"
      />
    </div>
    <div
      className="flex-100 layout-row layout-align-start-center"
    >
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            Pickup Date
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            28-03-2018
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            ETD 
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            28-03-2018
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            ETA  
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            28-03-2018
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             Estimated T/T 
            <sup>
              *
            </sup>
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            0
             Days
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-30 layout-row layout-wrap layout-align-center"
  >
    <div
      className="flex-90 layout-row layout-align-space-between-center layout-wrap"
    >
      <p
        className="flex-none"
      >
        Total price: 
      </p>
      <Price
        currency="EUR"
        scale=""
        value={112}
      />
    </div>
    <div
      className="flex-90 layout-row layout-align-space-between-center layout-wrap"
    >
      <RoundButton
        active={true}
        back={false}
        disabled={false}
        handleDisabled={null}
        handleNext={[Function]}
        icon=""
        iconClass=""
        inverse={false}
        size="full"
        text="Choose"
        theme={
          Object {
            "colors": Object {
              "primary": "#333",
              "secondary": "#fafafa",
            },
          }
        }
      />
    </div>
  </div>
</div>
`;
